controllers/snapshot/snapshot_adapter.go

```diff
@@ -147,6 +153,100 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (reconciler.Operation
 	return reconciler.ContinueProcessing()
 }
 
+// EnsureCreationOfEnvironment makes sure that all envrionments that were requested via
+// IntegrationTestScenarios get created, in case that environment is already created, provides
+// a message about this fact
+func (a *Adapter) EnsureCreationOfEnvironment() (reconciler.OperationResult, error) {
+	if gitops.HaveHACBSTestsFinished(a.snapshot) {
+		a.logger.Info("The Snapshot has finished testing.")
+		return reconciler.ContinueProcessing()
+	}
+
+	integrationTestScenarios, err := helpers.GetAllIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+
+	if err != nil {
+		a.logger.Error(err, "Failed to get Integration test scenarios for the following application",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+		return reconciler.RequeueOnErrorOrStop(err)
+	}
+
+	allEnvironments, err := a.getAllEnvironments()
+	if err != nil {
+		a.logger.Error(err, "Failed to get all environments.",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+		return reconciler.RequeueOnErrorOrStop(err)
+	}
+
+	environmentFound := false
+	if integrationTestScenarios != nil {
```

Comment:
Can we do

```go
if integrationTestScenarios == nil {
   a.logger.Info("No integration tests for snapshot .....")
   return reconciler.ContinueProcessing()
}
```

right after `GetAllIntegrationTestScenariosForApplication` we don't need environments without integrationScenarios

Reply:
Yes, added. Thanks.